src/main.c

```diff
@@ -836,12 +842,26 @@ static struct listener_config_t *find_listener(struct sockaddr *addr, socklen_t
     return NULL;
 }
 
-static struct listener_config_t *add_listener(int fd, struct sockaddr *addr, socklen_t addrlen, int is_global, int proxy_protocol)
+static struct listener_config_t *add_listener(int fd, int domain, int so_reuseport, struct sockaddr *addr, socklen_t addrlen,
+                                              int is_global, int proxy_protocol)
```

Comment:
As stated above, I do not think that we need to change the arguments of the function.

`listener->reuseport_fds` could be set to NULL (assuming that its type would be changed to `int *`), and can be modified in the caller of this function laterwards if `so-reuseport` was set to `ON`.

Reply:
fixed.